sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -0,0 +1,621 @@
+            topic_name = kwargs.get("topic_name")
+            subscription_name = kwargs.get("subscription_name")
+            if queue_name and topic_name:
+                raise ValueError("Queue/Topic name can not be specified simultaneously.")
+            if not (queue_name or topic_name):
+                raise ValueError("Queue/Topic name is missing. Please specify queue_name/topic_name.")
+            if topic_name and not subscription_name:
+                raise ValueError("Subscription name is missing for the topic. Please specify subscription_name.")
+
+            entity_name = queue_name or topic_name
+
+            super(ServiceBusReceiver, self).__init__(
+                fully_qualified_namespace=fully_qualified_namespace,
+                credential=credential,
+                entity_name=entity_name,
+                **kwargs
+            )
+        self._message_iter = None
+        self._create_attribute(**kwargs)
+        self._connection = kwargs.get("connection")
+        self._session = ServiceBusSession(self._session_id, self, self._config.encoding) if self._session_id else None
+        self._prefetch = kwargs.get("prefetch")
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self._can_run()
+        while True:
+            try:
+                return self._do_retryable_operation(self._iter_next)
+            except StopIteration:
+                self.close()
+                raise
+
+    next = __next__  # for python2.7
+
+    def _iter_next(self):
+        self._open()
+        uamqp_message = next(self._message_iter)
+        message = self._build_message(uamqp_message)
+        return message
+
+    def _create_handler(self, auth):
+        self._handler = ReceiveClient(
+            self._get_source_for_session_entity() if self._session_id else self._entity_uri,
+            auth=auth,
+            debug=self._config.logging_enable,
+            properties=self._properties,
+            error_policy=self._error_policy,
+            client_name=self._name,
+            on_attach=self._on_attach_for_session_entity if self._session_id else None,
+            auto_complete=False,
+            encoding=self._config.encoding,
+            receive_settle_mode=self._mode.value,
+            send_settle_mode=SenderSettleMode.Settled if self._mode == ReceiveSettleMode.ReceiveAndDelete else None,
+            timeout=self._config.idle_timeout * 1000 if self._config.idle_timeout else 0,
+            prefetch=self._config.prefetch
+        )
+
+    def _open(self):
+        if self._running:
+            return
+        if self._handler:
+            self._handler.close()
+
+        auth = None if self._connection else create_authentication(self)
+        self._create_handler(auth)
+        self._handler.open(connection=self._connection)
+        self._message_iter = self._handler.receive_messages_iter()
+        while not self._handler.client_ready():
```

Comment:
Should this have some sort of circuit breaker/failure stop point wherein we give up?  I worry it could incur looping forever in certain situations, but this may be a moot point depending on how client_ready works.

Reply:
moved the topic to here: https://github.com/Azure/azure-sdk-for-python/issues/10602